core/src/main/java/org/apache/gravitino/job/local/LocalJobExecutor.java

```diff
@@ -255,15 +255,18 @@ public void runJob(Pair<String, JobTemplate> jobPair) {
     try {
       String jobId = jobPair.getLeft();
       JobTemplate jobTemplate = jobPair.getRight();
+
+      Process process;
       synchronized (lock) {
         jobStatus.put(jobId, Pair.of(JobHandle.Status.STARTED, UNEXPIRED_TIME_IN_MS));
-      }
 
-      LocalProcessBuilder processBuilder = LocalProcessBuilder.create(jobTemplate, configs);
-      Process process = processBuilder.start();
-      runningProcesses.put(jobId, process);
+        LocalProcessBuilder processBuilder = LocalProcessBuilder.create(jobTemplate, configs);
+        process = processBuilder.start();
+        runningProcesses.put(jobId, process);
```

Comment:
So, what is the root cause of CI failure? Based on your changes, the real process and status changes are not within a lock.

Reply:
There's a problem if the `cancel` happens before the process start, it will lead to a wrong status.

Reply:
I see, good fix.